### DISCUSSION TITLE
How to use pre-installed MSYS2 on vcpkg?

### DISCUSSION THREAD

**Discussion by brunvonlope** [microsoft/vcpkg#42039] [Q&A] [answered]
Some packages requires msys2 binaries at build time so they download them

But I have msys2 installed at `C:/msys64`. It is possible to configure vcpkg to use it so avoid downloading?

**Comment by dg0yt** [comment, marked as answer]
There is no way to do this at the moment.

It is not trivial.
- It is unknown if the external msys root has all required packages for a port. Otherwise there might be unexpected buid errors.
- The build might accidentally pick dependencies from the external msys root, e.g. via pkg-config. There might be unexpected errors. Maybe very subtle ones, occuring on runtime.
- Mixing different versions of msys2 (or cygwin) runtime is known to cause errors.
